chasm/ref.go

```diff
@@ -41,18 +41,22 @@ type EntityKey struct {
 type ComponentRef struct {
 	EntityKey
 
-	// This is needed for routing
-	rootComponentName string
-	// or maybe make it specific
+	// Fully qualified component type name for routing.
+	rootComponentType string
+	// We could also make the routing infomation clear with
 	// routingKey string
 
+	// We can also simply put shardID here.
+	// TODO: Remove one of shardID and rootComponentType.
 	shardID int32
 
-	// From the component name, we can find the component struct definition
-	// use reflection to find sub-components and understand if those sub-components
-	// needs to be loaded or not. we only need to do this for sub-components of the component.
-	// path for parent/ancenstor component can be inferred from the path.
-	componentName      string
+	// Fully qualified component type name.
+	// From the component type, we can find the component struct definition,
+	// then use reflection to find sub-components and understand if those sub-components
+	// need to be loaded or not.
+	// We only need to do this for sub-components, path for parent/ancenstor components
+	// can be inferred from the current component path.
+	componentType      string
```

Comment:
I am a little in doubt between `componentType` and `componentTypeName`. I am ok to use the former for now, and see if it is confusing. We can rename it once again later.